src/core/net/icmp6.hpp

```diff
@@ -72,13 +72,21 @@ class IcmpHeader: public otIcmp6Header
      * ICMPv6 Message Types
      *
      */
-    typedef otIcmp6Type Type;
+    enum Type
+    {
+        kTypeDstUnreach  = 1,     ///< Destination Unreachable
+        kTypeEchoRequest = 128,   ///< Echo Request
+        kTypeEchoReply   = 129,   ///< Echo Reply
```

Comment:
Suggestion: can we define these to use the `OT_ ICMP6` constants? (so we have only place where the actual value is given).
```
enum Type
{
     kTypeDstUnreach  = OT_ICMP6_TYPE_DST_UNREACH,     ///< Destination Unreachable
     ...